system-x/services/azure/service-bus/src/main/java/software/tnb/azure/service/bus/validation/ServiceBusValidation.java

```diff
@@ -0,0 +1,76 @@
+package software.tnb.azure.service.bus.validation;
+
+import software.tnb.azure.common.account.AzureServiceBusAccount;
+import software.tnb.azure.service.bus.service.ErrorProcessor;
+import software.tnb.azure.service.bus.service.MessageProcessor;
+
+import com.azure.messaging.servicebus.ServiceBusClientBuilder;
+import com.azure.messaging.servicebus.ServiceBusMessage;
+import com.azure.messaging.servicebus.ServiceBusProcessorClient;
+import com.azure.messaging.servicebus.ServiceBusSenderClient;
+import com.azure.messaging.servicebus.administration.ServiceBusAdministrationClient;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ServiceBusValidation {
+    private final AzureServiceBusAccount azureServiceBusAccount;
+    private final ServiceBusAdministrationClient adminClient;
+
+    public ServiceBusValidation(AzureServiceBusAccount azureServiceBusAccount, ServiceBusAdministrationClient adminClient) {
+        this.azureServiceBusAccount = azureServiceBusAccount;
+        this.adminClient = adminClient;
+    }
+
+    public AzureServiceBusAccount getAzureServiceBusAccount() {
+        return azureServiceBusAccount;
+    }
+
+    public void createQueue(String queue) {
+        adminClient.createQueue(queue);
+    }
+
+    public void deleteQueue(String queue) throws InterruptedException {
+        Thread.sleep(4000L);
+        adminClient.deleteQueue(queue);
+    }
+
+    public void sendMessage(String queue, String message) {
+        ServiceBusSenderClient client = new ServiceBusClientBuilder()
+            .connectionString(azureServiceBusAccount.connectionString())
+            .sender()
+            .queueName(queue)
+            .buildClient();
+
+        client.sendMessage(new ServiceBusMessage(message));
+    }
+
+    public List<String> receiveMessages(String queue) {
+        MessageProcessor messageProcessor = new MessageProcessor();
+        ErrorProcessor errorProcessor = new ErrorProcessor();
+
+        final ServiceBusProcessorClient client = new ServiceBusClientBuilder()
+            .connectionString(azureServiceBusAccount.connectionString())
+            .processor()
+            .queueName(queue)
+            .processMessage(messageProcessor)
+            .processError(errorProcessor)
+            .buildProcessorClient();
+
+        try (client) {
+            client.start();
+            //WaitUtils.waitFor(waitUntilSupplier, retries, timeout, "no messages received");
+            Thread.sleep(4000L);
+            client.stop();
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+
+        if (!(errorProcessor.getErrors().isEmpty())) {
+            return errorProcessor.getErrors();
+        } else {
+            return messageProcessor.getMessages().stream().map(serviceBusReceivedMessage -> serviceBusReceivedMessage.getBody().toString()).collect(
+                Collectors.toList());
+        }
+    }
```

Comment:
better to use waiter than thread sleep, the waiter should exit once the client receives or a message or an error, something like:

```
public List<String> receiveMessages(final String queue) {
        final MessageProcessor messageProcessor = new MessageProcessor();
        final ErrorProcessor errorProcessor = new ErrorProcessor();

        try (ServiceBusProcessorClient client = new ServiceBusClientBuilder()
            .connectionString(azureServiceBusAccount.connectionString())
            .processor()
            .queueName(queue)
            .processMessage(messageProcessor)
            .processError(errorProcessor)
            .buildProcessorClient()) {

            client.start();
            WaitUtils.waitFor(() -> !(errorProcessor.getErrors().isEmpty()
                && messageProcessor.getMessages().isEmpty())
                ,10, 1000, "No messages received");
        }

       return Stream.concat(errorProcessor.getErrors().stream()
            , messageProcessor.getMessages().stream()
            .map(serviceBusReceivedMessage -> serviceBusReceivedMessage.getBody().toString()))
            .collect(Collectors.toList());
    }
```
since `ServiceBusProcessorClient` implements `java.lang.AutoCloseable` you don't need to call `client.stop()` in a try-with-resource statement

moreover if you expects one of the two returned list empty (error or messages) you can concatenate the streams with `Stream.concat` to avoid `if` statement

Reply:
You are right, this looks better